### PR TITLE
separate mocha node.js definition

### LINKED ISSUE #4701: mocha.d.ts should not depend on node.d.ts
Mocha works on both browsers and node -- it *should* not be bringing in node definition files for those of us that use it only on the browser. This is causing us problems when we try to update it.

### PR DESCRIPTION
**Pull request by vvakame**
fix #4701

### CODE REVIEW

**Comment by vvakame** [comment]
@enlight @thasner @adidahiya can I merge this PR?

**Comment by adidahiya** [comment]
:+1:

File: mocha/mocha-node.d.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Type definitions for mocha 2.2.5
```
Comment (enlight) [inline mocha/mocha-node.d.ts]: Could you add a note here to make it obvious why there are two separate versions of the typings? Perhaps something like:
```TypeScript
// Type definitions for mocha 2.2.5 (running on node.js)
```
And in `mocha.d.ts`
```TypeScript
// Type definitions for mocha 2.2.5 (running in a browser)
```

Comment (adidahiya) [inline mocha/mocha-node.d.ts]: Making the comment more specific sounds fine, but that kind of convention should be standardized across DefinitelyTyped and therefore self-evident from the file names.

**Comment by enlight** [comment]
I don't have access to a machine I can test these changes on right now, but I don't see any reason they wouldn't work for me, so :+1: from me. Looks like [custom reporters aren't supported when running Mocha in a browser](https://github.com/mochajs/mocha/issues/1592) anyway.

**Comment by thasner** [comment]
:+1:

**Comment by vvakame** [comment]
:+1:

**Pull request merged**